ext/iconv/iconv.c

```diff
@@ -2670,8 +2670,14 @@ static int php_iconv_stream_filter_append_bucket(
 							}
 							pd = out_buf;
 						} else {
+							char *tbuf;
+							if (NULL == (tbuf = pestrndup(out_buf, (out_buf_size - ocnt), persistent))) {
+								goto out_failure;
+							}
+
 							if (NULL == (new_out_buf = perealloc(out_buf, new_out_buf_size, persistent))) {
```

Comment:
This is actually dead code: perealloc is infallible, it aborts/bails out if the allocation fails. As such, the code in the if condition can simply be dropped. (Same for the check of pemalloc above.)